common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -270,9 +272,17 @@ protected Path getRecoveryPath() {
     }
 
     if (_recoveryPath == null) {
-      _recoveryPath = new Path(localDirs[0]);
+      for (String dir : localDirs) {
+        try {
+          DiskChecker.checkDir(new File(dir));
+          _recoveryPath = new Path(dir);
+          return _recoveryPath;
+        } catch (DiskErrorException e) {
+          logger.warn("Local disk {} not available", dir, e);
+        }
+      }
     }
 
-    return _recoveryPath;
+    return null;
```

Comment:
I would throw an exception to let it fail if the code reaches to here, which means all the disks are unavailable.